runtime/storage.go

```diff
@@ -61,344 +54,241 @@ func NewStorage(
 	return &Storage{
 		Ledger:                ledger,
 		PersistentSlabStorage: persistentSlabStorage,
-		writes:                map[interpreter.StorageKey]atree.Storable{},
-		readCache:             map[interpreter.StorageKey]atree.Storable{},
-		contractUpdates:       map[interpreter.StorageKey]atree.Storable{},
-		reportMetric:          reportMetric,
+		writes:                map[interpreter.StorageKey]atree.StorageIndex{},
+		storageMaps:           map[interpreter.StorageKey]*interpreter.StorageMap{},
+		contractUpdates:       map[interpreter.StorageKey]*interpreter.CompositeValue{},
 	}
 }
 
-// ValueExists returns true if a value exists in account storage.
-//
-func (s *Storage) ValueExists(
-	_ *interpreter.Interpreter,
-	address common.Address,
-	key string,
-) bool {
+const storageIndexLength = 8
 
-	storageKey := interpreter.StorageKey{
+func (s *Storage) GetStorageMap(address common.Address, domain string) (storageMap *interpreter.StorageMap) {
+	key := interpreter.StorageKey{
 		Address: address,
-		Key:     key,
+		Key:     domain,
 	}
 
-	// Check locally
+	storageMap = s.storageMaps[key]
```

Comment:
Should we gracefully handle the non-existing keys?
```go
storageMap, ok := s.storageMaps[key]
```

Reply:
The value type of `storageMaps` is `*interpreter.StorageMap`, so Go will use the default value for the type here when the lookup fails, which is `nil`, so this should already be safe (and is tested)